src/shogun2-core/src/main/java/de/terrestris/shogun2/service/Csv2ExtJsLocaleService.java

```diff
@@ -0,0 +1,144 @@
+package de.terrestris.shogun2.service;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.elasticsearch.tools.content.StructureUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Service;
+
+import com.opencsv.CSVReader;
+
+/**
+ *
+ * @author Nils Bühner
+ *
+ */
+@Service("localeService")
+public class Csv2ExtJsLocaleService {
+
+	@Autowired
+	private ResourceLoader resourceLoader;
+
+	/**
+	 *
+	 * @param locale
+	 * @param locale2
+	 * @return
+	 * @throws Exception
+	 */
+	@SuppressWarnings("unchecked")
+	public Map<String, Object> getAllComponentsForLocale(String appId, String locale) throws Exception {
+
+		Map<String, Object> resultMap = new TreeMap<String, Object>();
+		Resource csvResource = resourceLoader.getResource("classpath:META-INF/locale/" + appId + ".csv"); // TODO make location configurable
+
+		if (!csvResource.exists()) {
+			throw new Exception("CSV locale resource for " + appId + " does not exist.");
+		}
+
+		Reader reader = new FileReader(csvResource.getFile());
+		CSVReader csvReader = new CSVReader(reader, ';', '"', '\\');
+
+		try {
+			int columnIndexOfLocale = detectColumnIndexOfLocale(locale, csvReader);
+
+			List<String> nextLine;
+			while ((nextLine = Arrays.asList(ArrayUtils.nullToEmpty(csvReader.readNext()))).isEmpty() == false) {
```

Comment:
No. I am not kidding as the csvReader does not provide a `hasNext()` method... :wink: 
But i'll try to simplify this in another way...